### PR TITLE
Fix tensor detach warnings triggered during example tests.

### PR DESCRIPTION
**Pull request by drivanov**
Follow-up to [PR#10357](https://github.com/pyg-team/pytorch_geometric/pull/10357) and [PR#10360](https://github.com/pyg-team/pytorch_geometric/pull/10360) to address and resolve the warnings similar to the following ones:
```
/workspace/examples/gat.py:66: UserWarning: Converting a tensor with requires_grad=True to a scalar may lead to unexpected behavior.
Consider using tensor.detach() first. (Triggered internally at /opt/pytorch/pytorch/torch/csrc/autograd/generated/python_variable_methods.cpp:835.)
  return float(loss)

/workspace/examples/proteins_gmt.py:68: UserWarning: Converting a tensor with requires_grad=True to a scalar may lead to unexpected behavior.
Consider using tensor.detach() first. (Triggered internally at /opt/pytorch/pytorch/torch/csrc/autograd/generated/python_variable_methods.cpp:835.)
  total_loss += data.num_graphs * float(loss)
  ```

### CODE REVIEW

**Review by puririshi98 — APPROVED** [review]
LGTM
